### ISSUE TITLE
Upgrade to numba 0.39

### ISSUE THREAD

**Issue by Juanlu001** [poliastro/poliastro#406]
numba 0.39 was released yesterday and it gained support for `ndarray.dot`:

https://groups.google.com/a/continuum.io/d/msg/numba-users/fsAiSQ39ztg/YSfidKXJAAAJ

which will for jitting code that uses `v.dot(v)` to retain units as well.

**Comment by Juanlu001** [comment]
However, bear in mind that a jitted function won't preserve units:

```python
In [5]: norm([1, 2, 3] * u.km)
Out[5]: <Quantity 3.74165739 km>

In [6]: jit(norm)([1, 2, 3] * u.km)
Out[6]: 3.7416573867739413
```

**Comment by Juanlu001** [comment]
This has been fixed in #400.
